### PR TITLE
Feat/cli label indices

### LINKED ISSUE #231: feat: Command line flag to print jumpdest offset/pc of all jump labels
## Overview
Self explanatory, a command line flag to print the resolved jump labels.

### PR DESCRIPTION
**Pull request by merklefruit**
## Overview

Adds a `huff_cli` flag to print the jump labels and their respective PC offset. Works for the `MAIN` macro.

Closes #231 

## Usage Example

```shell
cargo run ./huff-examples/erc20/contracts/ERC20.huff --label-indices
```

Output:

```text
------ JUMP LABEL INDICES ------
  allowance: 0x18e
  approve: 0x177
  balanceOf: 0x14e
  error: 0xd8
  is_owner: 0xed
  mints: 0xde
  totalSupply: 0x16b
  transfer: 0x48
--------------------------------
```

## Notes / open questions

- The output format (what gets printed by the command) could be better, I'm not sure if there is a standard I should follow, looking for feedback on this
- Is support for the `MAIN` macro sufficient or should this command include all macros?

### CODE REVIEW

File: huff_cli/src/huffc.rs [hunk 1]
```diff
@@ -204,6 +208,60 @@ fn main() {
...
+                if contracts.len() > 1 {
+                    eprintln!(
+                        "{}",
+                        Paint::red("Multiple contracts found. Please specify a single contract and try again.")
+                    );
+                    std::process::exit(1);
+                }
+
+                if let Some(contract) = contracts.first() {
+                    let m_macro = contract.find_macro_by_name("MAIN").unwrap();
```
Comment (cheethas) [inline huff_cli/src/huffc.rs:223]: There is an alternative macro cli flag you can use here to support any macro

File: huff_cli/src/huffc.rs [hunk 2]
```diff
@@ -204,6 +208,60 @@ fn main() {
...
+                    // Format the label indices nicely
+                    let label_indices = bytecode_res
+                        .label_indices
+                        .iter()
+                        .map(|(label, index)| format!("{}: {:#04x}", label, index))
+                        .collect::<Vec<_>>()
+                        .join("\n  ");
+
+                    println!("\n------ JUMP LABEL INDICES ------");
+                    println!("  {}", label_indices);
```
Comment (cheethas) [inline huff_cli/src/huffc.rs:246]: With regards to the formatting comment, there is a super lib comfy table that was used in the `huffc test` report by @clabby.
Link to usage in other parts of the codebase here: https://github.com/huff-language/huff-rs/blob/94c34e402a46365fee29863aa08558af178c2b94/huff_tests/src/report.rs#L2

File: huff_cli/src/huffc.rs [hunk 3]
```diff
@@ -82,6 +82,10 @@ struct Huff {
     #[clap(short = 'v', long = "verbose")]
     verbose: bool,
 
+    /// Prints out the jump label PC indices for the specified contract.
+    #[clap(short = 'x', long = "label-indices")]
```
Comment (cheethas) [inline huff_cli/src/huffc.rs:86]: hmm i see that `-l` is taken by alt constructor. `-l` suits this one waaaaay better. Maybe swap alt-constructor to `-t`? Open to suggestions though its not a widely used feature.

**Review by cheethas — COMMENTED** [review]
This looks like really nice work. Thank you for doing this ser!!!!!

File: huff_cli/src/huffc.rs (hunk 2 above)
Comment (merklefruit) [reply huff_cli/src/huffc.rs:246]: oh wow, that library is amazing! I didn't know it existed :)

File: huff_cli/src/huffc.rs (hunk 1 above)
Comment (merklefruit) [reply huff_cli/src/huffc.rs:223]: added support for alt-main

File: huff_cli/src/huffc.rs (hunk 3 above)
Comment (merklefruit) [reply huff_cli/src/huffc.rs:86]: changed it to `-l` as suggested for now

File: huff_cli/src/huffc.rs (hunk 2 above)
Comment (merklefruit) [reply huff_cli/src/huffc.rs]: Looking way better:

Screenshot 2023-04-16 at 01 12 22

**Review by cheethas — APPROVED** [review]
sorry for the delay for the re-review! This looks great!

**Pull request merged**